pandas/core/generic.py

```diff
@@ -3888,6 +3888,8 @@ def drop(
 
         for axis, labels in axes.items():
             if labels is not None:
+                if isinstance(labels, (list, Index)) and len(labels) == 0:
```

Comment:
Should probably use `is_list_like(labels)` here.

Reply:
if so, then also test with ndarray?  others?

Reply:
use not len(labels)

Reply:
Changed in the new commit.